### PR TITLE
Update networking APIs

### PR DESCRIPTION
**Pull request by kyessenov**
- clarify short name resolution
- remove faults (for now)

Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: networking/v1alpha3/destination_rule.proto [hunk 1]
```diff
@@ -72,14 +72,14 @@ message DestinationRule {
...
-  // For example on Kubernetes, when hosts contains a short name, Istio
-  // will interpret the short name based on the namespace of the client
-  // where rules are being applied. Thus, when a client in the "default"
-  // namespace applies a rule containing a name "reviews", Istio will setup
-  // routes to the "reviews.default.svc.cluster.local" service. However, if
-  // a different name such as "reviews.sales" is used, it would be treated
-  // as a FQDN during virtual host matching.  In Consul, a plain service
-  // name would be resolved to the FQDN "reviews.service.consul".
+  // For example on Kubernetes, when hosts contains a short name, Istio will
+  // interpret the short name based on the namespace of the rule.  Thus, when a
```
Comment (ZackButcher) [inline networking/v1alpha3/destination_rule.proto]: Remove double spaces here and elsewhere please.

File: networking/v1alpha3/virtual_service.proto [hunk 2]
```diff
@@ -271,16 +271,14 @@ message Destination {
...
-  // an external service, such that the sidecar will dynamically resolve
-  // the DNS of the service name and route the request to the IP addresses
+  // platform specific manner.  For example in Kubernetes, when a route with a
+  // short name "reviews" in the destination in namespace "bookinfo" is applied
+  // to a client in , the final destination is resolved to
+  // reviews.bookinfo.svc.cluster.local. The sidecar will route to the IP
+  // addresses of the pods constituting the service. However, if the lookup
+  // fails, "reviews" is treated as an external service, such that the sidecar
+  // will dynamically resolve the DNS of the service name and route the request
+  // to the IP addresses
```
Comment (ZackButcher) [inline networking/v1alpha3/virtual_service.proto]: nit: bring `returned by DNS.` up to this line, it looks funky atm.

File: networking/v1alpha3/virtual_service.proto [hunk 3]
```diff
@@ -341,6 +339,7 @@ message HTTPRoute {
   // Retry policy for HTTP requests.
   HTTPRetry retries = 7;
 
+  // $hide_from_docs
```
Comment (ZackButcher) [inline networking/v1alpha3/virtual_service.proto]: Why remove the fault injection fields?

Comment (kyessenov) [reply networking/v1alpha3/virtual_service.proto]: Issue #388

Comment (ZackButcher) [reply networking/v1alpha3/virtual_service.proto]: Thanks

File: networking/v1alpha3/destination_rule.proto (hunk 1 above)
Comment (kyessenov) [reply networking/v1alpha3/destination_rule.proto]: Done

File: networking/v1alpha3/virtual_service.proto (hunk 2 above)
Comment (kyessenov) [reply networking/v1alpha3/virtual_service.proto]: Done

**Comment by rshriram** [comment]
While you are here, could you also update the blurb about destination subsets? The fact that if they don't exist, traffic will blackhole, and that the recommended configuration is to have a default catch all route that sends traffic to an existing destination? This should be all docs changes only.

**Comment by kyessenov** [comment]
@rshriram can you point me to where it says something about subset defaulting? for now, it's unspecified, so it's ok to blackhole. We'll need more APIs to clarify what to do anyways.

File: authentication/v1alpha1/istio.authentication.v1alpha1.pb.html [hunk 4]
```diff
@@ -608,17 +608,14 @@ <h2 id="istio.networking.v1alpha3.Destination">istio.networking.v1alpha3.Destina
...
-destination as &ldquo;reviews.sales&rdquo;, the resolution process first looks for
-a &ldquo;reviews&rdquo; service in the &ldquo;sales&rdquo; namespace. In both cases, the
-sidecar will route to the IP addresses of the pods constituting the
-service. However, if the lookup fails, &ldquo;reviews.sales&rdquo; is treated as
-an external service, such that the sidecar will dynamically resolve
-the DNS of the service name and route the request to the IP addresses
-returned by the DNS.</p>
+platform specific manner. For example in Kubernetes, when a route with a
+short name &ldquo;reviews&rdquo; in the destination in namespace &ldquo;bookinfo&rdquo; is applied
+to a client in , the final destination is resolved to
```
Comment (louiscryan) [inline authentication/v1alpha1/istio.authentication.v1alpha1.pb.html]: .. client in <what?> ...

**Comment by louiscryan** [comment]
LGTM

File: networking/v1alpha3/virtual_service.proto [hunk 5]
```diff
@@ -125,14 +125,14 @@ message VirtualService {
...
-  // will interpret the short name based on the namespace of the client
-  // where rules are being applied. Thus, when a client in the "default"
-  // namespace applies a rule containing a name "reviews, Istio will setup
-  // routes to the "reviews.default.svc.cluster.local" service. However, if
-  // a different name such as "reviews.sales" is used, it would be treated
-  // as a FQDN during virtual host matching. In Consul, a plain service
-  // name would be resolved to the FQDN "reviews.service.consul".
+  // For example on Kubernetes, when hosts contains a short name, Istio will
+  // interpret the short name based on the namespace of the rule. Thus, when a
+  // client namespace applies a rule in the "default" namespace containing a name
```
Comment (frankbu) [inline networking/v1alpha3/virtual_service.proto]: s/client namespace/client/

File: networking/v1alpha3/virtual_service.proto [hunk 6]
```diff
@@ -125,14 +125,14 @@ message VirtualService {
...
+  // "reviews, Istio will setup routes to the "reviews.default.svc.cluster.local"
```
Comment (frankbu) [inline networking/v1alpha3/virtual_service.proto]: missing quote: "reviews"

File: networking/v1alpha3/virtual_service.proto [hunk 7]
```diff
@@ -271,17 +271,14 @@ message Destination {
...
-  // destination as "reviews.sales", the resolution process first looks for
-  // a "reviews" service in the "sales" namespace. In both cases, the
-  // sidecar will route to the IP addresses of the pods constituting the
-  // service. However, if the lookup fails, "reviews.sales" is treated as
-  // an external service, such that the sidecar will dynamically resolve
-  // the DNS of the service name and route the request to the IP addresses
-  // returned by the DNS.
+  // platform specific manner. For example in Kubernetes, when a route with a
+  // short name "reviews" in the destination in namespace "bookinfo" is applied
+  // to a client in , the final destination is resolved to
```
Comment (frankbu) [inline networking/v1alpha3/virtual_service.proto]: // short name "reviews" in the destination of a rule in namespace "bookinfo" is applied,
// the final destination is resolved to

**Pull request merged**